spec/materialist/materializer_spec.rb

```diff
@@ -49,7 +49,14 @@ def actions_called
       end
 
       class << self
+        attr_accessor :error_to_throw_once_in_find_or_initialize_by
+
         def find_or_initialize_by(source_url:)
+          if(err = error_to_throw_once_in_find_or_initialize_by)
```

Comment:
I think the error would be thrown on `save`, not on `find_or_initialize_by`; but I guess it doesn't matter much here.

Reply:
good point, although it doens't matter but I will follow for clarity

Reply:
OK, on a second thought it's not very trivial to do the same on `save!`, it's not ideal, but maybe we can keep this as it is?

Reply:
Sure no big deal I think 👍